### ISSUE TITLE
Identify types of users on platform so we can build features for specific types.

### ISSUE THREAD

**Issue by kslottow** [pculture/unisubs#2651]
Currently we do not identify different types of users on Amara platform (AoD linguist, TED volunteer, PM, etc.).  We need this as a framework for AoD-specific features (eg. timesheet link, dashboard items, feedback forms, support portal).

**Comment by bendk** [comment]
Ignore that assignment stuff, just testing out github.

**Comment by nslottow** [comment]
This needs to be broken down into steps, or separate issues for the different features we want to implement to get this done.

Some amount of this can be automated by what we already know about users, but some amount will have to be voluntary on the part of users.

**Comment by kslottow** [comment]
@nslottow Yeah, we can make this into an epic and nest the component parts as issues inside it.

**Comment by kslottow** [comment]
@nslottow please list components I should look into for this, thanks!

**Comment by bendk** [comment]
I'm following this one by accident, so please ignore me if I'm missing something, but what about using the team homepage for this?  If AOD teams had those links on their dashboard, then it would be pretty simple for users to go there.  Not as quick as a menu item that was always there, but still fairly quick.  The nice thing about this way is it seems fairly clean with little guesswork or manual flagging of users.

**Comment by nslottow** [comment]
If that's a discoverable way to existing AoD teams to self-identify, then that seems fairly straightforward.  Questions I'd have about this to understand exactly what we're looking for (without looking at the existing code):

- Is self-identification by choosing from a list of available AoD/TED teams enough, or is there some authority that we would like to manually bestow/revoke team membership?
- Is there any kind of access control or authentication model we want to put in place?
- How would we like to automate this process when adding new users to teams (e.g. what does that UI look like, who can access it, and how)?
- Do we care about management or UI at all for this feature, or are we just trying to tag teams currently working for PCF to spin up the new team features?
- What are the main types of queries we're going to be doing on the data?  i.e. How will this information be used in the planned AoD specific features?

**Comment by bendk** [comment]
This makes me think that teams + standard django groups could work.  Possibly with a bit of customization.

For AoD:
  - The teams do not have open membership.  I think they are invite only, so I think it's fine to rely on membership.
  - You could use a django group to handle permissions.  This doesn't 100% solve the problem, since admins still would need to add the users to the group.  If we wanted it to be automatic, I still think trying to use the standard systems could make sense.  For example, maybe making it so that if you join a particular team then you automatically get added to a group.
  - Adding users to teams is pretty standard and we have good ways to do it.
  - I'm not really sure about the last 2 bullet points, I figured it was mostly about a) providing a way for users to go to certain pages and b) permissions checks on those pages.  Is there more complexity than that?

I don't really know about the TED side of things.  What functionality do they need besides there team pages?

**Comment by deanjansen** [comment]
As we look further into the possibilities, let's be sure we check in with AoD to get a gut-check on whatever process(es) we're considering. I have a feeling self-identification wouldn't be right for AoD members (though not 100% sure).

@kslottow do you think it would make sense to create some simple/concrete questions (based on this thread) and loop back with AoD for input?

**Comment by kslottow** [comment]
Hi guys!  Sorry for my delayed response.

@bendk identifying teams sounds like it would be very useful as a stepping stone to a couple of things that should be accessible to AoD linguists: support portal, timesheet link, and feedback forms (and possibly other things we think of later).  The only thing on my list that wouldn't be addressed by team identification would be a spot for a linguist to view his/her monthly payment details.  That's something I was hoping would go on an individual's dashboard, but maybe there's a way around it, totally open to suggestions.  If working on a team level is clean with little guesswork or manual flagging of users, I think it's a great option.  This may be an ignorant question, but how are individual AoD users able to get messages that only go to them?  Aren't they "tagged" in some way to enable the messaging system?

@nslottow On the last two bullet points (how do I make a bullet point?):
1. I was not thinking of this as having any UI component, if we don't need users to manually tag themselves (doing it the team way would eliminate that, do I understand that correctly?)  The UI would come in when we make the features that appear on each team's dashboard, like the support portal and timesheet links.
2. Thanks for the translation - "How will this information be used in the planned AoD specific features?" I love this question because we should definitely tag teams in a way that facilitates the features we want to make for them.  I hope this answers it, at least partially: the identifying info should allow us to make new features appear on the dashboards of only specific teams/groups*.  This will very likely include a timesheet and support portal link that appears on the dashboards of AoD teams and NOT volunteer teams.

@deanjansen Yes, I would like to do exactly that!  Let's go over that tomorrow :)

*I don't know what kinds of features will be most useful yet because I haven't looked into the needs of specific teams/groups

**Comment by kslottow** [comment]
@bendk @nslottow here's another way identifying teams would be used:
 #2646

**Comment by kslottow** [comment]
@nslottow @bendk @deanjansen 
It seems like getting into this without a more specific idea of the features that it will be used for and which user groups they will apply to would not be a smart use of resources.  Kind of like building the foundation of a house without knowing how big the house will be or what it will be made of.  Thanks for looking at this and laying it out for me to understand that!  I'll revisit this after some user discovery and as the AoD blueprints of processes come together (working on this with Dean and Aleli).
